### PR TITLE
Javalab: move commit modal subcomponents into their own files

### PR DESCRIPTION
**Pull request by bencodeorg**
Follow-up to https://github.com/code-dot-org/code-dot-org/pull/42268.

Splits commit modal into three files. The new files (`CommitDialogBody`, `CommitDialogFile`) represent components that already existed, but were stored in the `CommitDialog` file. No new code added in this PR, just moving existing code into separate files.

## Links

Jira: [CSA-791](https://codedotorg.atlassian.net/browse/CSA-791)

## Testing story

I manually tested that the Commit Dialog continued to work as expected (rendered / able to commit files).

### CODE REVIEW

File: apps/src/javalab/CommitDialogBody.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import color from '@cdo/apps/util/color';
+import CommitDialogFile from './CommitDialogFile';
+
+export const fileShape = {
+  name: PropTypes.string.isRequired,
+  commit: PropTypes.bool.isRequired,
+  hasConflictingName: PropTypes.bool.isRequired
+};
+
+const PADDING = 8;
```
Comment (bencodeorg) [inline apps/src/javalab/CommitDialogBody.jsx:13]: I could not for the life of me figure out how to import this constant from `CommitDialog`. It was `undefined` in `styles` object -- tried moving the `styles` declaration/assignment above the function declaration, but no luck. Some sort of hoisting thing I don't understand??

Comment (molly-moen) [reply apps/src/javalab/CommitDialogBody.jsx:13]: I think you would need to do `export const PADDING` in `CommitDialog`, but it's a pretty minor copy/paste so I don't feel strongly about it needing to happen

File: apps/src/javalab/CommitDialogFile.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import i18n from '@cdo/javalab/locale';
+import color from '@cdo/apps/util/color';
+import {fileShape} from './CommitDialogBody';
+
+const PADDING = 8;
+
+export default function CommitDialogFile({file, onToggleFile}) {
```
Comment (molly-moen) [inline apps/src/javalab/CommitDialogFile.jsx]: nit: now that this is in its own file, `CommitDialogFile` is a confusing name. Could we change to `CommitDialogFileSelector` or something?

File: apps/src/javalab/CommitDialogBody.jsx (hunk 1 above)
Comment (molly-moen) [reply apps/src/javalab/CommitDialogBody.jsx:13]: (git is being weird about my comment here so sorry if this is a duplicate): I think you need to do `export const PADDING = 8` and then you'll be able to import. It's a pretty minor duplicate so no worries if you can't make it work.

Comment (bencodeorg) [reply apps/src/javalab/CommitDialogBody.jsx:13]: I think I tried that, with an import statement like `import {PADDING} from ./CommitDialog` (does that look right) and just kept getting errors related to it being undefined. I guess I'll just leave it

Comment (sanchitmalhotra126) [reply apps/src/javalab/CommitDialogBody.jsx:13]: Just double checking here, looks like you have `const PADDING = 8` declared in `CommitDialog` but haven't exported it - does it work if you do `export const PADDING = 8`?

Comment (bencodeorg) [reply apps/src/javalab/CommitDialogBody.jsx:13]: It doesn't :(. Interestingly, when I did try that, I console logged the value of `styles` and `PADDING` in the component function, PADDING was set to 8 but the styles object had NaN in places where PADDING was used. ???

File: apps/src/javalab/CommitDialogFile.jsx (hunk 2 above)
Comment (bencodeorg) [reply apps/src/javalab/CommitDialogFile.jsx]: Made it `CommitDialogFileRow`, how does that sound?

**Review by molly-moen — APPROVED** [review]
LGTM!

**Pull request merged**